### PR TITLE
[Flutter Driver] Simplified the serialization/deserialization logic of the Descendant/…

### PR DESCRIPTION
**Pull request by adazh**
…Ancestor matchers.

Instead of flattening the "of" and "matching" matchers to the JSON string, using jsonEncode/jsonDecode to convert the matchers to String/from String.

This will make the serialization in Espresso/Java much simpler.

## Description
Updated the serialization/deserialization logic for the Descendant/Ancestor matchers. Take the "get_diagnostics_tree" for example.
Before:
{"id":"message-7","method":"ext.flutter.driver","params":{"isolateId":"isolates/4120307314110843",**"of_keyValueType":"String","of_keyValueString":"TwoTimesCounterContainer","of_finderType":"ByValueKey","matching_type":"Text", "matcher_finderType":
"ByType","finderType":"Descendant"**,"diagnosticsType":"widget","includeProperties":true,"subtreeDepth":20,"command":"get_diagnostics_tree","timeout":10000},"jsonrpc":"2.0"}

After:
{"id":"message-7","method":"ext.flutter.driver","params":{"isolateId":"isolates/4120307314110843",**"of":"{\\"keyValueType\\":\\"String\\",\\"keyValueString\\":\\"TwoTimesCounterContainer\\",\\"finderType\\":\\"ByValueKey\\"}","matching":"{
\\"type\\":\\"Text\\",\\"finderType
\":\\"ByType\\"}"**,"finderType":"Descendant","diagnosticsType":"widget","includeProperties":true,"subtreeDepth":20,"command":"get_diagnostics_tree","timeout":10000},"jsonrpc":"2.0"}

## Tests

flutter_test test/src/extension_test.dart still passes.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by goderbauer** [comment]
This didn't require any changes to tests?

**Comment by adazh** [comment]
Ah, forgot about the find_test.dart. On it.

**Comment by adazh** [comment]
> This didn't require any changes to tests?

Tests are updated.

File: packages/flutter_driver/lib/src/common/find.dart [hunk 1]
```diff
@@ -391,31 +385,23 @@ class Ancestor extends SerializableFinder {
...
-      } else if (key.startsWith('matching_')) {
-        matching[key.substring('matching_'.length)] = json[key];
-      } else {
-        other[key] = json[key];
-      }
-    }
+    final Map<String, String> jsonOfMatcher =
+        Map<String, String>.from(jsonDecode(json['of']));
+    final Map<String, String> jsonMatchingMatcher =
+    Map<String, String>.from(jsonDecode(json['matching']));
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/common/find.dart]: nit: indentation.

**Review by goderbauer — APPROVED** [review]
LGTM

Technically, this is a breaking change for anybody using the API,  but that's probably just flutter driver and you?

**Comment by adazh** [comment]
> LGTM
> 
> Technically, this is a breaking change for anybody using the API, but that's probably just flutter driver and you?

Yep, I think so.

**Pull request merged**